packages/core/src/components/private/BaseInputRange/index.tsx

```diff
@@ -0,0 +1,205 @@
+import Text from '../../Text';
+import { stylesheetInputRange, HANDLE_SIZE, HALF_HANDLE_SIZE, MARK_SIZE } from './styles';
+
+/**
+ * By default, the **edges** (not the _center_) of the slider handle align with
+ * the **edges** of the slider track. Thus the total width of the slider is not
+ * `100%` but `100% - HANDLE_SIZE_PX`. The default position is correct at 50%,
+ * so we offset based on a fractional distance from the center ()
+ * position based on the percent
+ */
+const getCorrectedPosition = memoize(
+  (value: number, min: number, max: number, width: number) => {
+    const valueRange = max - min;
+    const centerX = width / 2;
+    const pxPosition = ((value - min) / valueRange) * width;
+    // positive offset if value < center, negative offset if value > center
+    const pxFromCenter = centerX - pxPosition;
+    const fractionFromCenter = pxFromCenter / centerX;
+    const pxOffset = fractionFromCenter * HALF_HANDLE_SIZE;
+    const leftPosition = pxPosition + pxOffset;
+    return leftPosition;
+  },
+  (...args) => JSON.stringify(args),
+);
+
+export type BaseInputRangeProps = {
+  /** Whether to always show a tooltip with value. */
+  alwaysShowTooltip?: boolean;
+  /** Any values to annotate. */
+  annotations?: { value: number; label?: string }[];
+  /** Whether to disable the input. */
+  disabled?: boolean;
+  /** Unique id of the input. */
+  id: string;
+  /** Max range value. */
+  max?: number;
+  /** Min range value. */
+  min?: number;
+  /** Callback invoked on input value change. */
+  onChange: (value: number, event: React.ChangeEvent<HTMLInputElement>) => void;
+  /** Override rendering of tooltip content. */
+  renderTooltipContent?: (value: number) => React.ReactNode;
+  /** Whether to show a tooltip with value on hover. */
+  showTooltip?: boolean;
+  /** Step size for the range. */
+  step?: number;
+  /** Current value. */
+  value?: number;
+  /** Width of the input. */
+  width?: number;
+};
+
+type BaseInputRangeState = {
+  showPopup: boolean;
+};
+
+class BaseInputRange extends React.Component<
+  BaseInputRangeProps & WithStylesProps,
+  BaseInputRangeState
+> {
+  static defaultProps = {
+    renderTooltipContent: (value: number) => value.toFixed(0),
+  };
+
+  state = {
+    showPopup: !!this.props.alwaysShowTooltip,
+  };
+
+  private handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.props.onChange(Number(event.currentTarget.value), event);
+  };
+
+  private handleMouseEnter = () => {
+    if (this.props.showTooltip && !this.props.alwaysShowTooltip) {
+      this.setState({ showPopup: true });
+    }
+  };
+
+  private handleMouseLeave = () => {
+    if (this.props.showTooltip && !this.props.alwaysShowTooltip) {
+      this.setState({ showPopup: false });
+    }
+  };
+
+  renderTooltip(leftOffset: number) {
+    const { renderTooltipContent, value = 0, cx, styles } = this.props;
+    return (
+      <div
+        role="tooltip"
```

Comment:
I didn't think this needed the complexity / perf implications of the `Tooltip` component so implemented a simplified version (open to updating based on what you think). 

The designs have the tooltip on **top**, but I implemented on the **bottom** because if `renderTooltipContent` results in `>1` line of text, the fixed positioning on top would be wrong without more complex logic / dynamic positioning (used in `Tooltip`) 🤷‍♀️

Reply:
i think this tooltip should match the other tooltip's default state (white background, normal text) and have the option to `invert`